packages/backend/src/models/ProjectModel/ProjectModel.ts

```diff
@@ -569,57 +570,125 @@ export class ProjectModel {
+                let exploreCache = await this.database(CachedExploreTableName)
+                    .select('explore')
+                    .whereRaw('? = ANY(table_names)', tableName)
+                    .andWhere('project_uuid', projectUuid)
+                    .first();
+
+                span.setAttribute(
+                    'foundIndividualExploreCache',
+                    !!exploreCache,
+                );
+                if (!exploreCache) {
+                    // fallback: check all cached explores
+                    // try finding explore via base table name
+                    exploreCache = await this.getExploreQueryBuilder(
+                        projectUuid,
+                    ).andWhereRaw("explore->>'baseTable' = ?", [tableName]);
+
+                    if (!exploreCache) {
+                        // try finding explore via joined table alias
+                        // Note: there is an edge case where we return the wrong explore because join table aliases are not unique
+                        exploreCache = await this.getExploreQueryBuilder(
+                            projectUuid,
+                        ).andWhereRaw(
+                            "(explore->>'tables')::json->? IS NOT NULL",
+                            [tableName],
+                        );
+                    }
+                }
 
-        return explore
-            ? ProjectModel.convertMetricFiltersFieldIdsToFieldRef(
-                  explore.explore,
-              )
-            : undefined;
+                return exploreCache
+                    ? ProjectModel.convertMetricFiltersFieldIdsToFieldRef(
+                          exploreCache.explore,
+                      )
+                    : undefined;
+            },
+        );
     }
 
     async saveExploresToCache(
         projectUuid: string,
         explores: (Explore | ExploreError)[],
     ): Promise<DbCachedExplores> {
-        const [cachedExplores] = await this.database(CachedExploresTableName)
-            .insert({
-                project_uuid: projectUuid,
-                explores: JSON.stringify(explores),
-            })
-            .onConflict('project_uuid')
-            .merge()
-            .returning('*');
-        return cachedExplores;
+        return wrapOtelSpan(
+            'ProjectModel.saveExploresToCache',
+            {},
+            async () => {
+                // delete previous individually cached explores
+                await this.database(CachedExploreTableName)
+                    .where('project_uuid', projectUuid)
+                    .delete();
+                // cache explores individually
+                await this.database(CachedExploreTableName)
+                    .insert(
+                        explores.map((explore) => ({
+                            project_uuid: projectUuid,
+                            name: explore.name,
+                            table_names: Object.keys(explore.tables || {}),
```

Comment:
```suggestion
                            table_names: Object.keys(explore.tables || []),
```
Database is storing  `Text[]` should this be [] instead of {} ?

Reply:
this value is inside `Object.keys()`